### PR TITLE
Bach: Refactor Series sorting behavior

### PR DESCRIPTION
**Pull request by KathiaBarahona**
Removed `Series.index_sorting` and `Series.sorted_ascending` properties. Now it relies only in `order_by` property as we do in `DataFrame`

### CODE REVIEW

File: bach/bach/operations/concat.py [hunk 1]
```diff
@@ -321,15 +321,19 @@ def _get_concatenated_object(self) -> Series:
...
+            base_node=node,
             name=final_result_series.name,
             expression=Expression.column_reference(final_result_series.name),
-            index={} if self.ignore_index else main_series.index,
+            index={} if self.ignore_index else index,
             group_by=None,
-            sorted_ascending=None,
-            index_sorting=[] if self.ignore_index else main_series.index_sorting,
+            order_by=[],
             instance_dtype=series_cls.dtype  # TODO: make this work for structural types too
```
Comment (KathiaBarahona) [inline bach/bach/operations/concat.py:337]: `Series.__init__` is not validating if indexes have same node as parent Series. Opened issue: #1036

File: bach/bach/series/series.py [hunk 2]
```diff
@@ -538,18 +507,13 @@ def copy_override(
         Special case:
         * If index is not None, then index_sorting is automatically set to `[]` unless overridden
```
Comment (KathiaBarahona) [inline bach/bach/series/series.py:508]: I'll remove this

File: bach/bach/series/series.py [hunk 3]
```diff
@@ -125,8 +125,6 @@ def __init__(
         name: str,
         expression: Expression,
         group_by: Optional['GroupBy'],
-        sorted_ascending: Optional[bool],
-        index_sorting: List[bool],
         instance_dtype: StructuredDtype,
         order_by: Optional[List[SortColumn]] = None,
```
Comment (thijs-obj) [inline bach/bach/series/series.py:129]: request: could we make this non-optional? I have the feeling that having this optional just hides places where we forget to call this correctly.

```suggestion
        order_by: List[SortColumn],
```

Additionally, this will save like a line of code in this function

File: bach/bach/series/series.py [hunk 4]
```diff
@@ -748,10 +714,16 @@ def __set_item_with_merge(self, other: 'Series') -> Tuple['Series', 'Series']:
             # incorrect second expression = a + b - b
             # correct second expression = a + b__other - b
             caller_expr = self.expression.replace_column_references(other.name, f'{other.name}__other')
-            caller_series = self.copy_override(base_node=df.base_node, expression=caller_expr)
+            caller_series = self.copy_override(
```
Comment (thijs-obj) [inline bach/bach/series/series.py:717]: Question: Is this an unrelated fix?

File: bach/bach/series/series.py [hunk 5]
```diff
@@ -932,25 +901,14 @@ def to_frame(self) -> DataFrame:
 
         The DataFrame returned has the grouping and sorting also set like this Series had.
         """
-        if self._sorted_ascending is not None:
```
Comment (thijs-obj) [inline bach/bach/series/series.py:935]: nice to get this cleaned up :ok

**Review by thijs-obj — CHANGES_REQUESTED** [review]
When searching for `index_sorting` I find 8 occurrences. `sorted_ascending` gives 3 results. From the looks of it all of those need to be changed.

Nice to get this cleaned up!

File: bach/bach/series/series.py (hunk 4 above)
Comment (KathiaBarahona) [reply bach/bach/series/series.py:717]: Related to issue https://github.com/objectiv/objectiv-analytics/issues/1036. Since now we base_nodes when setting sort, it was raising an error

File: bach/bach/series/series.py (hunk 3 above)
Comment (KathiaBarahona) [reply bach/bach/series/series.py:129]: mmm has the same typing as `DataFrame.__init__`. Was aiming more for consistency.

**Comment by KathiaBarahona** [comment]
> When searching for `index_sorting` I find 8 occurrences. `sorted_ascending` gives 3 results. From the looks of it all of those need to be changed.
> 
> Nice to get this cleaned up!

Thanks for checking! Missed a couple of them, fixed here https://github.com/objectiv/objectiv-analytics/pull/1037/commits/433dd32f24d3483112e5007076cc25f3931bf84c

File: bach/bach/series/series.py (hunk 3 above)
Comment (KathiaBarahona) [reply bach/bach/series/series.py:129]: Matching what the docstring says  https://github.com/objectiv/objectiv-analytics/blob/17c87cea3ea87cca0dcb6842a7d6b77d30025ee3/bach/bach/dataframe.py#L192

Comment (thijs-obj) [reply bach/bach/series/series.py:129]: okay I think we disagree. I just think it is slightly better, because it saves a bit of code and is less ambiguous between functions. You feel different, that's fine. Let's go with the changes you have made.

File: bach/bach/series/series.py (hunk 4 above)
Comment (thijs-obj) [reply bach/bach/series/series.py:714]: ack :+1:

**Review by thijs-obj — APPROVED** [review]
